pkgs/development/python-modules/wavefile/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, setuptools
+, pyaudio
+, numpy
+, libsndfile
+}:
+
+buildPythonPackage rec {
+  pname = "wavefile";
+  version = "1.5";
+
+  src = fetchFromGitHub {
+    owner = "vokimon";
+    repo = "python-wavefile";
+    rev = "python-wavefile-${version}";
+    sha256 = "9sHj1gb93mCVpejRGSdLJzeFDCeTflZctE7kMWfqFrE=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  buildInputs = [
+    pyaudio
+    libsndfile
+  ];
+
+  propagatedBuildInputs = [
+    numpy
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+    pyaudio
+    numpy
+    libsndfile
+  ];
+
+  # Fix check error
+  # OSError: libsndfile.so.1: cannot open shared object file: No such file or directory
+  LD_LIBRARY_PATH = lib.makeLibraryPath [ libsndfile ];
```

Comment:
This is the problem, in `$out//lib/python3.9/site-packages/wavefile/libsndfile.py`:

```
_lib=None
try:
    from ctypes.util import find_library
    #does the user already have libsamplerate installed?
    if sys.platform == 'win32' :
        dllPath = find_library(dllName)
    else :
        dllPath = dllName
    _lib = ct.CDLL(dllPath)
except:
    try:
        #if not, get the dll installed with the wrapper
        import os
        dllPath = os.path.dirname(os.path.abspath(__file__))
        _lib = ct.CDLL(os.path.join(dllPath, dllName))
    except:
        raise Exception("could not import libsndfile dll, make sure the dll '%s' is in the path"%(dllName))
```

this should be replaced with:

```
_lib = ct.CDLL("${lib.getLib libsndfile}/lib/libsndfile.${stdenv.hostPlatform.extensions.sharedLibrary}")